qubesfiletrust/qvm_file_trust.py

```diff
@@ -0,0 +1,546 @@
+                        help='Print all local folders considered untrusted')
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='Do not print to stdout')
+
+    # Only require a path for certain options
+    if not '--printfolders' in sys.argv and not '-p' in sys.argv:
+        parser.add_argument('paths', metavar='path',
+                            type=str, nargs='+', help='a folder or file path')
+
+    args = parser.parse_args()
+
+    # Set global quiet variable based on given flags
+    global OUTPUT_QUIET
+    OUTPUT_QUIET = args.quiet
+
+    # Grab and cache untrusted phrase
+    global UNTRUSTED_PHRASE
+    try:
+        with open(PHRASE_FILE_LOC) as phrase_file:
+            for line in phrase_file.readlines():
+                line = line.rstrip()
+
+                # Ignore comments
+                if not line.startswith('#'):
+                    UNTRUSTED_PHRASE = line
+                    break
+
+    except:
+        serror('Unable to open phrase file: {}'.
+                format(PHRASE_FILE_LOC))
+
+    # Error checking
+    if args.trusted and args.untrusted:
+        error('--trusted and --untrusted options cannot both be set')
+        sys.exit(64)
+    if args.check and (args.trusted or args.untrusted):
+        error('--trusted or --untrusted '
+              'cannot be set while --is-trusted is set')
+        sys.exit(64)
+    if args.check_multiple and args.check_multiple_all_untrusted:
+        error('--check_multiple and --check_multiple_all_untrusted '
+              'options cannot both be set')
+        sys.exit(64)
+
+    if args.printfolders:
+        print_folders()
+        return
+
+    checking_multiple = args.check_multiple or \
+                        args.check_multiple_all_untrusted
+
+    # Determine which action to take for each given path
+    for path in args.paths:
+        if not (args.check or args.trusted or args.untrusted) \
+            or args.check:
+            if (not args.check_multiple_all_untrusted and \
+                not args.check_multiple) and len(args.paths) > 1:
+                error('Use --check-multiple to check multiple paths')
+                sys.exit(64)
+            if os.path.isdir(path):
+                # Check folder
+                check_folder(path, checking_multiple)
+            elif not os.path.isdir(path):
+                # Check file
+                check_file(path, checking_multiple)
+
+        elif os.path.isdir(path):
+            if args.trusted:
+                # Set folder as trusted
+                change_folder(path, True)
+            elif args.untrusted:
+                # Set folder as untrusted
+                change_folder(path, False)
+
+        elif not os.path.isdir(path):
+            if args.trusted:
+                # Set file as trusted
+                change_file(path, True)
+            elif args.untrusted:
+                # Set file as untrusted
+                change_file(path, False)
```

Comment:
Changing directory to untrusted should at least mark all files inside as untrusted. But probably also kick the daemon to reload configuration (or emit a warning about it).

Reply:
Yeah, daemon should watch the rule lists for changes and act upon them. 

I think we originally planned to have the daemon handle recursively marking untrusted folders, and it will mark it permanently. Would be bad if it wasn't running though. Think we shouldn't rely on it in that case then?

Reply:
Safer option would be to mark files here, then notify the daemon to mark future files in those directories.

Reply:
Good point, will mark them then.